apiserver/facades/client/bundle/bundle.go

```diff
@@ -150,11 +160,149 @@ func (b *BundleAPI) GetChanges(args params.BundleChangesParams) (params.BundleCh
 	return results, nil
 }
 
+var fail = func(failErr error) (params.StringResult, error) {
+	return params.StringResult{}, common.ServerError(failErr)
+}
+
 // ExportBundle exports the current model configuration as bundle.
 func (b *BundleAPI) ExportBundle() (params.StringResult, error) {
-	return params.StringResult{}, nil
+	if err := b.checkCanRead(); err != nil {
+		return fail(err)
+	}
+
+	var exportConfig state.ExportConfig
+
+	b.backend.SetExportconfig(exportConfig)
+	logger.Criticalf("XXXXXXXXXXXXXXXXX...........SkipStatusHistory: %v ", exportConfig.SkipStatusHistory)
+	model, err := b.backend.ExportPartial(exportConfig)
+	if err != nil {
+		return fail(err)
+	}
+
+	// Fill it in charm.BundleData datastructure.
+	bundleData, err := b.FillBundleData(model)
+	if err != nil {
+		return fail(err)
+	}
+
+	bytes, err := yaml.Marshal(bundleData)
+	if err != nil {
+		return fail(err)
+	}
+
+	return params.StringResult{
+		Result: string(bytes),
+	}, nil
 }
 
 // Mask the new method from V1 API.
 // ExportBundle is not in V1 API.
 func (u *APIv1) ExportBundle() (_, _ struct{}) { return }
+
+// FillBundle fills the bundledata datastructure required for the exportBundle.
+func (b *BundleAPI) FillBundleData(model description.Model) (*charm.BundleData, error) {
+	cfg := model.Config()
+	var defaultSeries string
```

Comment:
This may leave defaultSeries uninitialised...

Something like this 
defaultSeries, ok := cfg["default-series"]
if !ok {
 // should never happen, but just in case it does
// let's set it to latest lts
}

and the for-loop becomes redundant... no need to iterate over the whole of config if we are only after one value.